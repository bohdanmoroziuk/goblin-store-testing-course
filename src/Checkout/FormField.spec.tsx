import React from "react";
import { render, fireEvent } from "@testing-library/react";

import { FormField } from "./FormField";

describe("FormField", () => {
  it("renders correctly", () => {
    const { getByTestId, container } = render(
      <FormField label="Foo label" name="foo" />
    );

    const input = getByTestId("input");

    expect(container).toMatchSnapshot();
    expect(input).toBeInTheDocument();
    expect(input).not.toHaveClass("is-error");
    expect(input).toHaveAttribute("name", "foo");
  });
  
  describe("with error", () => {
    it("renders error message", () => {
      const { getByText } = render(
        <FormField
          label="Foo label"
          name="foo"
          errors={{ message: "Example error" }}
        />
      );

      expect(getByText("Example error")).toBeInTheDocument();
    });
  });

  describe("on change", () => {
    it("normalizes the input", () => {
      const { getByTestId } = render(
        <FormField
          label="Foo label"
          name="foo"
          normalize={(value: string) => value.toUpperCase()}
        />
      );

      const input = getByTestId("input") as HTMLInputElement;

      fireEvent.change(input, { target: { value: "test" }});

      expect(input.value).toBe("TEST");
    });
  });
});