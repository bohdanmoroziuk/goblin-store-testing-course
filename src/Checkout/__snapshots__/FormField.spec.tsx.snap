// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormField renders correctly 1`] = `
<div>
  <div
    class="nes-field"
  >
    <label
      for="foo"
    >
      Foo label
      :
    </label>
    <input
      class="nes-input undefined"
      data-testid="input"
      name="foo"
    />
  </div>
</div>
`;
